cmd/call-api-endpoint/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"strings"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/edulinq/autograder/internal/api"
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/cmd"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/log"
+)
+
+var args struct {
+	config.ConfigArgs
+	cmd.CommonOptions
+
+	Endpoint   string   `help:"Endpoint of the desired API." arg:""`
+	Parameters []string `help:"Parameter(s) for the endpoint" arg:"" optional:""`
+}
+
+func main() {
+	kong.Parse(&args,
+		kong.Description("Execute an API request to the specified endpoint. For more information on available API endpoints, see the API resource file at: ../resources/api.json"),
+	)
+
+	err := config.HandleConfigArgs(args.ConfigArgs)
+	if err != nil {
+		log.Fatal("Failed to load config options.", err)
+	}
+
+	var description *core.EndpointDescription
+
+	apiDescription := api.Describe(*api.GetRoutes())
+	for endpoint, requestResponse := range apiDescription.Endpoints {
+		if endpoint == args.Endpoint {
+			description = &requestResponse
+			break
+		}
+	}
+
+	if description == nil {
+		log.Fatal("Failed to find the endpoint.", log.NewAttr("endpoint", args.Endpoint))
+
+		// Return after log.Fatal() sets the exit code to 1 to avoid proceeding during tests.
+		return
+	}
+
+	request := map[string]any{}
+	for _, arg := range args.Parameters {
+		// Split the parameter into it's key and value.
+		parts := strings.SplitN(arg, ":", 2)
+		if len(parts) != 2 {
+			log.Fatal("Invalid parameter format: missing a colon. Expected format is 'key:value'. E.g. target-email:bob@test.edulinq.org", log.NewAttr("parameter", parts))
```

Comment:
s/. E.g./, e.g., /g
"e.g." is typically followed be a comma (and usually doesn't start sentences).

s/target-email:bob@test.edulinq.org/'target-email:bob@test.edulinq.org'
Since you quote the template ('key:value'), quote the example.

s/$/./
Try for complete sentences.